website/pages/docs/theming/scale-out-theme.mdx

```diff
@@ -0,0 +1,519 @@
+
+export default typography
+```
+
+### 💨 Transition
+You can customize transition like this:
+
+```ts live=false
+// transition.ts
+
+const transitionProperty = {
+  common:
+    "background-color, border-color, color, fill, stroke, opacity, box-shadow, transform",
+  colors: "background-color, border-color, color, fill, stroke",
+  dimensions: "width, height",
+  position: "left, right, top, bottom",
+  background: "background-color, background-image, background-position",
+}
+
+const transitionTimingFunction = {
+  "linear": "cubic-bezier(0, 0, 1, 1)",
+  "ease-in": "cubic-bezier(0.42, 0, 1, 1)",
+  "ease-out": "cubic-bezier(0, 0, 0.58, 1)",
+  "ease-in-out": "cubic-bezier(0.42, 0, 0.58, 1)",
+  // More...
+}
+
+const transitionDuration = {
+  "custom-speed": "250ms",
+  // More ...
+}
+
+const transition = {
+  property: transitionProperty,
+  easing: transitionTimingFunction,
+  duration: transitionDuration,
+}
+
+export default transition
+```
+
+
+### 🗂 Z-index
+We recommend trying to solve stacking issues without `z-index`.
+You can learn more about [stacking contexts and z-index here](/guides/z-index).
+
+```ts live=false
+const zIndices = {
+  headerNavigation: 1750,
+  // More...
+}
+
+export type ZIndices = typeof zIndices
+
+export default zIndices
+```
+
+### 📦 Export Foundations
+Export all above overwritten Foundations in `index.ts`. Put this file under `/foundations` folder.
+
+```ts live=false
+// index.ts
+
+import borders from "./borders"
+import breakpoints from "./breakpoints"
+import colors from "./colors"
+import radii from "./radius"
+import shadows from "./shadows"
+import sizes from "./sizes"
+import { spacing } from "./spacing"
+import transition from "./transition"
+import typography from "./typography"
+import zIndices from "./z-index"
+
+const foundations = {
+  breakpoints,
+  zIndices,
+  radii,
+  colors,
+  ...typography,
+  sizes,
+  shadows,
+  space: spacing,
+  borders,
+  transition,
+}
+
+type FoundationsType = typeof foundations
+export interface Foundations extends FoundationsType {}
+
+export default foundations
+```
+
+---
+
+## 📁 Components
+
+<ComponentLinks
+  github={{ package: "theme/src/components/" }}
+/>
+
+In this folder, we define the style that will be the default for the components.
+
+We will learn [how to define the Component Styles in the next section](/docs/theming/component-style).
+For that reason, we only provides a simple example for splitting Component Styles into files.
+
+
+### 🛎 Button Component Styles
+We can define `<Button>` component styles like this:
+
+```ts live=false
+// button.ts
+
+// Styles for the base style
+const baseStyle = {
+  fontWeight: 'bold',
+  textTransform: 'uppercase',
+}
+
+// Styles for the size variations
+const sizes = {
+  sm: {
+    fontSize: '12px',
+    padding: '16px',
+  },
+  md: {
+    fontSize: '16px',
+    padding: '24px',
+  },
+}
+
+// Styles for the visual style variations
+const variants = {
+  outline: {
```

Comment:
I think we could add an example how to use the `mode(lightModeValue, darkModeValue)(props)` function here.

https://github.com/chakra-ui/chakra-ui/blob/327e1624d22936abb43068e1f57054e43c9c6819/packages/theme/src/components/button.ts#L52